src/cogent3/core/new_alignment.py

```diff
@@ -2374,3 +2379,559 @@ def _(
+        seqid: str,
+        start: OptInt = None,
+        stop: OptInt = None,
+    ) -> numpy.ndarray:
+        """Return sequence corresponding to seqid as an array of indices.
+        start/stop are in alignment coordinates. Includes gaps.
+        """
+        seq = self.seqs[seqid]
+        gaps = self.gaps[seqid]
+
+        # if there's no gaps, just slice the sequence
+        if len(seq) == self.align_len:
+            return seq[start:stop]
+
+        return self.moltype.most_degen_alphabet().gapped_by_map(seq, gaps)[start:stop]
+
+    def get_seq_str(
+        self,
+        *,
+        seqid: str,
+        start: OptInt = None,
+        stop: OptInt = None,
+    ) -> str:
+        """Return sequence corresponding to seqid as a string.
+        start/stop are in alignment coordinates. Excludes gaps."""
+
+        return self.alphabet.from_indices(
+            self.get_seq_array(seqid=seqid, start=start, stop=stop)
+        )
+
+    def get_gapped_seq_str(
+        self,
+        *,
+        seqid: str,
+        start: OptInt = None,
+        stop: OptInt = None,
+    ) -> str:
+        """Return sequence corresponding to seqid as a string.
+        start/stop are in alignment coordinates. Includes gaps."""
+
+        return self.alphabet.from_indices(
+            self.get_gapped_seq_array(seqid=seqid, start=start, stop=stop)
+        )
+
+    def get_seq_bytes(
+        self,
+        *,
+        seqid: str,
+        start: OptInt = None,
+        stop: OptInt = None,
+    ) -> bytes:
+        """Return sequence corresponding to seqid as a bytes string.
+        start/stop are in alignment coordinates. Excludes gaps."""
+        return self.get_seq_str(seqid=seqid, start=start, stop=stop).encode("utf8")
+
+    def get_gapped_seq_bytes(
+        self,
+        *,
+        seqid: str,
+        start: OptInt = None,
+        stop: OptInt = None,
+    ) -> bytes:
+        """Return sequence corresponding to seqid as a bytes string.
+        start/stop are in alignment coordinates. Includes gaps."""
+        return self.get_gapped_seq_str(seqid=seqid, start=start, stop=stop).encode(
+            "utf8"
+        )
+
+    def add_seqs(
+        self, seqs: dict[str, StrORArray], force_unique_keys=True
+    ) -> AlignedData:
+        """Returns a new AlignedData object with added sequences."""
+        if force_unique_keys and any(name in self.names for name in seqs):
+            raise ValueError("One or more sequence names already exist in collection")
+        new_data = {
+            **self.seqs,
+            **seqs,
+        }
+        return self.__class__.from_aligned_seqs(new_data, self.moltype)
+
+    def reversed(self):
+        # todo: kath
+        ...
+
+    def subset(self):
+        # todo: kath
+        ...
+
+    def to_alphabet(self):
+        # todo: kath
+        ...
+
+    def to_rich_dict(self):
+        # todo: kath
+        ...
+
+
+class AlignedDataView(new_sequence.SeqViewABC, new_sequence.SliceRecordABC):
```

Comment:
Add the dunder methods for strings, bytes and arrays to this view class. (I note here that these dunder methods should be made abstract methods on `new_sequence. SeqViewABC`).